packages/aws-cdk-lib/aws-sns/lib/subscription-filter.ts

```diff
@@ -209,6 +209,13 @@ export class SubscriptionFilter {
     return new SubscriptionFilter([{ exists: true }]);
   }
 
+  /**
+   * Returns a subscription filter for absence of attribute key.
+   */
+  public static notExistsFilter() {
+    return new SubscriptionFilter([{ exists: false }]);
+  }
```

Comment:
IMO the proposed solution in the original issue is better: https://github.com/aws/aws-cdk/issues/34707. It maintains backward compatibility as well.

```
  /**
   * Returns a subscription filter for attribute key matching.
   */
  public static existsFilter(existsCondition: boolean = true) {
    return new SubscriptionFilter([{ exists: existsCondition }]);
  }
  ```

Is there a rationale for changing the approach?

Reply:
See comment https://github.com/aws/aws-cdk/issues/34707#issuecomment-2977052510